### ISSUE TITLE
unknown lint: clippy

### ISSUE THREAD

**Issue by Dushistov** [rust-lang/rust-clippy#2501]
Here is reduced test case:

[foo.zip](https://github.com/rust-lang-nursery/rust-clippy/files/1777969/foo.zip)

In real code I has generated code, this was simulated by `src/foo-include.rs`.
I want disable warnings for this generated code, so my `src/foo.rs` looks like this:
```Rust
#![allow(clippy)]

include!("foo-include.rs");
```

Without `tests` subdirectory this works fine:
```sh
$ cargo +nightly clippy
lib: foo
   Compiling foo v0.1.0 (file:///tmp/foo)
    Finished dev [unoptimized + debuginfo] target(s) in 0.11 secs
```

but with `tests` subdirectory `cargo clippy` show warning:
```sh
$ cargo +nightly clippy
lib: foo
   Compiling foo v0.1.0 (file:///tmp/foo)
    Finished dev [unoptimized + debuginfo] target(s) in 0.11 secs
test: test_foo
   Compiling foo v0.1.0 (file:///tmp/foo)
warning: unknown lint: `clippy`
 --> src/foo.rs:1:10
  |
1 | #![allow(clippy)]
  |          ^^^^^^
  |
  = note: #[warn(unknown_lints)] on by default

    Finished dev [unoptimized + debuginfo] target(s) in 0.61 secs

```

**Comment by bemeurer** [comment]
Anyone has a solution to this warning?

**Comment by mikerite** [comment]
Try `#[cfg_attr(feature = "cargo-clippy", allow(clippy))]`

**Comment by sshashank124** [comment]
Now it's recommended to use
